src/utils/contractLogs.ts

```diff
@@ -0,0 +1,130 @@
+import type { EtherSwap } from "boltz-core/typechain/EtherSwap";
+import { Result, Signer, solidityPackedKeccak256 } from "ethers";
+import log from "loglevel";
+
+import { AssetType, RBTC } from "../consts/Assets";
+import { weiToSatoshi } from "./rootstock";
+
+const scanInterval = 10_000;
+
+export type LogRefundData = {
+    asset: AssetType;
+    blockNumber: number;
+    transactionHash: string;
+
+    preimageHash: string;
+    amount: bigint;
+    claimAddress: string;
+    refundAddress: string;
+    timelock: bigint;
+};
+
+export const getLogsFromReceipt = async (
+    signer: Signer,
+    etherSwap: EtherSwap,
+    txHash: string,
+): Promise<LogRefundData> => {
+    const receipt = await signer.provider.getTransactionReceipt(txHash);
+
+    for (const event of receipt.logs) {
+        if (
+            event.topics[0] !== etherSwap.interface.getEvent("Lockup").topicHash
+        ) {
+            continue;
+        }
+
+        return parseLockupEvent(etherSwap, event).data;
+    }
+
+    throw "could not find event";
+};
+
+async function* scanLogsForPossibleRefunds(
+    abortSignal: AbortSignal,
+    signer: Signer,
+    etherSwap: EtherSwap,
+) {
+    const [signerAddress, latestBlock] = await Promise.all([
+        signer.getAddress(),
+        signer.provider.getBlockNumber(),
+    ]);
+    log.info(`Scanning for possible refunds of: ${signerAddress}`);
+
+    const filter = etherSwap.filters.Lockup(null, null, null, signerAddress);
+
+    for (let toBlock = latestBlock; toBlock >= 0; toBlock -= scanInterval) {
```

Comment:
should we also add our deployBlock check here so it won't scan below that?

Reply:
Good idea